### PR TITLE
 Refactoring according to Siddhi modifications

### PR DESCRIPTION
**Pull request by sujanan**

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/execution/string/substrexecutors/Type1Executor.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.execution.string.substrexecutors;
+
+/**
+ * str:substr(<string sourceText> , <int beginIndex>).
+ */
+public class Type1Executor extends SubstrExecutor<Integer, Object> {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/execution/string/substrexecutors/Type1Executor.java]: Can we use meaningful names like indexBasedExecutor. If naming like that is not straight-forward improve class level comment to say what each Executor handle

File: component/src/main/java/org/wso2/extension/siddhi/execution/string/GroupConcatFunctionExtension.java [hunk 2]
```diff
@@ -91,168 +95,201 @@
...
+        return executor.getReturnType() == type;
+    }
+
+    private boolean isConstantAttribute(ExpressionExecutor executor) {
+        return executor instanceof ConstantExpressionExecutor;
     }
 
     @Override
-    public Object processAdd(Object o) {
+    public Object processAdd(Object o, ExtensionState extensionState) {
```
[the remark this answers is not in the document]
Comment (sujanan) [reply component/src/main/java/org/wso2/extension/siddhi/execution/string/GroupConcatFunctionExtension.java]: Fix with aef3060

File: component/src/main/java/org/wso2/extension/siddhi/execution/string/GroupConcatFunctionExtension.java [hunk 3]
```diff
@@ -91,168 +95,201 @@
...
+            canDestroy = true;
+        }
+
+        private void makeUnDestroyable() {
+            canDestroy = false;
+        }
+
+        @Override
+        public boolean canDestroy() {
+            return canDestroy;
```
[the remark this answers is not in the document]
Comment (sujanan) [reply component/src/main/java/org/wso2/extension/siddhi/execution/string/GroupConcatFunctionExtension.java]: Fix with aef3060

File: component/src/main/java/org/wso2/extension/siddhi/execution/string/substrexecutors/Type1Executor.java (hunk 1 above)
Comment (sujanan) [reply component/src/main/java/org/wso2/extension/siddhi/execution/string/substrexecutors/Type1Executor.java]: Fix with aef3060

**Pull request merged**